vignettes/speedup_options.Rmd

```diff
@@ -0,0 +1,458 @@
+        subset_draws(variable = "R")
+    }
+  } else {
+    NA
+  }
+})
+# CRPS for the Rt estimates
+rt_crps <- lapply(
+  Rt_estimated,
+  calc_crps,
+  truth = R
+)
+
+# Function to extract infection estimates
+infections_estimated <- lapply(results, function(x) {
+  if (is.null(x$error)) {
+    obj <- x$result$estimates$fit
+    if (inherits(obj, "stanfit")) {
+      extract(obj, "infections")$infections
+    } else {
+      obj |>
+        as_draws_matrix() |>
+        subset_draws(variable = "infections")
+    }
+  } else {
+    NA
+  }
+})
+
+# CRPS for the infections estimates
+infections_crps <- lapply(
+  infections_estimated,
+  calc_crps,
+  truth = infections_true
+)
+```
+
+We will now post-process the CRPS results to make them easier to visualise by adding a  "time" column and reshaping the data to long format.
+
+```{r fit_postprocessing}
+# Post-processing the CRPS results of the Rt estimates
+rt_df <- as.data.table(rt_crps)
+rt_df[, time := 1:.N]
+rt_df <- melt(
+  rt_df,
+  id.vars = c("time"),
+  variable.name = "model"
+)
+
+# Post-processing the CRPS results of the infection estimates
+infections_df <- as.data.table(infections_crps)
+infections_df[, time := 1:.N]
+infections_df <- melt(
+  infections_df,
+  id.vars = c("time"),
+  variable.name = "model"
+)
+```
+
+Let's visualise the CRPS results for the $R_t$ and infection estimates.
+
+```{r rt_plot}
+rt_plot <- ggplot(rt_df, aes(x = time, y = value, colour = model)) +
+  geom_line() +
+  scale_colour_brewer("Model", palette = "Dark2") +
+  scale_y_continuous(labels = label_number_auto()) +
+  labs(x = "Time", y = "CRPS", title = "Estimating Rt with various model options") +
+  ggplot2::theme_bw()
+plot(rt_plot)
+```
+
+```{r infections_plot}
+infections_plot <- ggplot(infections_df, aes(x = time, y = value, colour = model)) +
+  geom_line() +
+  scale_colour_brewer("Model", palette = "Dark2") +
+  scale_y_continuous(labels = label_comma()) +
+  labs(x = "Time", y = "CRPS", title = "Estimating infections with various model options") +
+  ggplot2::theme_bw()
+plot(infections_plot)
+```
+
+## Some considerations when using the model options
+
+### Mechanistic vs non-mechanistic models
+
+- Estimation in _EpiNow2_ using the mechanistic approaches (prior on $R_t$) is often much slower than the non-mechanistic approach. The mechanistic model is slower because it models intricate details about the processes and mechanisms that drive $R_t$ estimates using the renewal equation. The non-mechanistic model, on the other hand, runs much faster but does not use the renewal equation to generate infections. Because of this none of the options defining the behaviour of the reproduction number are available in this case, limiting flexibility. It also means that the model is questionable for forecasting. If performing a retrospective analysis, the _non-mechanistic model_ (where `rt = NULL`) is better suited and a faster alternative to the mechanistic $R_t$ model.
```

Comment:
```suggestion
- Estimation in _EpiNow2_ using the mechanistic approaches (prior on $R_t$) is often much slower than the non-mechanistic approach. The mechanistic model is slower because it models details about the processes and mechanisms that drive $R_t$ estimates using the renewal equation. The non-mechanistic model, on the other hand, runs much faster but does not use the renewal equation to generate infections. Because of this none of the options defining the behaviour of the reproduction number are available in this case, limiting flexibility. It also means that the model is questionable for forecasting. If performing a retrospective analysis, the _non-mechanistic model_ (where `rt = NULL`) is better suited and a faster alternative to the mechanistic $R_t$ model.
```

Reply:
Thanks